ccp/lib/clients/git/client.py

```diff
@@ -8,34 +8,28 @@
 
 class GitClient(CmdClient):
 
-    def __init__(self, git_url, git_branch, clone_location="/tmp/git_clone"):
+    def __init__(self, git_url, git_branch, clone_location="/var/index/repo"):
         super(GitClient, self).__init__(base_command="git")
         self.git_url = git_url
         self.git_branch = git_branch
         self.clone_location = clone_location
-        self.actual_clone_location = path.join(self.clone_location +
-                                               ''.join(
-                                                   random.choice(
-                                                       string.ascii_uppercase +
-                                                       string.digits
-                                                   ) for _ in range(6)))
 
     def cleanup_clone(self):
-        if path.exists(self.actual_clone_location):
-            rmdir(self.actual_clone_location)
+        if path.exists(self.clone_location):
```

Comment:
This could have very dangerous consequences. The reason i setup actual clone location to be randomized is to ensure multiple requests don't come in each others way. 

Imagine 2 users requesting some information at same time, what if one request does a checkout  (and the index has been modded in this time) while another request is still reading the data, will the checkout now not fail as the reading user has lock on the files? 

Also, you can replace rmtree(gc.actual_clone_location) with gc.cleanup_clone()

Reply:
we are not planning to do git clone per request. it will be cloned only once. and pulled in case of update

Reply:
There can still be interference will a git pull succeed if file handlers are open on one of the requests?